connectors/src/connectors/slack/auto_read_channel.ts

```diff
@@ -68,12 +77,71 @@ export async function autoReadChannel(
     if (joinChannelRes.isErr()) {
       return joinChannelRes;
     }
-    await SlackChannel.create({
+    const createdChannel = await SlackChannel.create({
       connectorId,
       slackChannelId,
       slackChannelName: remoteChannelName,
       permission: "read_write",
     });
+
+    const baseSearchUrl = `${DUST_FRONT_API}/api/v1/w/${connector.workspaceId}/data_source_views/search`;
+    const searchUrl =
+      baseSearchUrl +
+      `?kind=custom&vaultKind=global&dataSourceId=${connector.dataSourceId}`;
+    const searchRes = await fetch(searchUrl, {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${connector.workspaceAPIKey}`,
+      },
+    });
+
+    if (!searchRes.ok) {
+      logger.error({
+        connectorId,
+        channelId: slackChannelId,
+        searchUrl,
+        error: await searchRes.text(),
+      });
+      return new Err(new Error("Failed to join Slack channel in Dust."));
+    }
+
+    const dataSourceViews = (
```

Comment:
nit : you can destructure here to simplify the tests after :
```suggestion
    const [dataSourceView] = (
```